### PR TITLE
nixos/python: add pytest-bdd module.

### PR DESCRIPTION
**Pull request by jm2dev**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/python-modules/pytest-bdd/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1v5dazwmsz2rvhbwg1i4y9qnn0ch1yzflca9c7n93sq2f2603xiy";
+  };
+
+  propagatedBuildInputs = [ glob2 Mako parse parse-type py pytest six ];
+
+  # Tests require extra dependencies
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/pytest-bdd/default.nix]: that fine... just add them to checkInputs
```suggestion
  checkInputs = [ mock ];
```

**Comment by jonringer** [comment]
also, to adhere to contributing guidelines, please rename the PR and commit to:
```
pythonPackages.pytest-bdd: init at 3.2.0
```

**Comment by jonringer** [comment]
```
nix-env failed:
error: while evaluating anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-1-lassulus.ewr1.nix.ci/.gc-of-borg-outpaths.nix:44:12, called from undefined position:
while evaluating anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-1-lassulus.ewr1.nix.ci/pkgs/top-level/release-lib.nix:121:6, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-1-lassulus.ewr1.nix.ci/lib/attrsets.nix:292:43:
while evaluating 'hydraJob' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-1-lassulus.ewr1.nix.ci/lib/customisation.nix:166:14, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-1-lassulus.ewr1.nix.ci/pkgs/top-level/release-lib.nix:121:14:
while evaluating the attribute 'meta' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-1-lassulus.ewr1.nix.ci/lib/customisation.nix:171:24:
while evaluating the attribute 'maintainers' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-1-lassulus.ewr1.nix.ci/pkgs/development/python-modules/pytest-bdd/default.nix:21:5:
undefined variable 'jm2dev' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-1-lassulus.ewr1.nix.ci/pkgs/development/python-modules/pytest-bdd/default.nix:21:39
```
you need to add yourself to maintainers, please look at https://github.com/NixOS/nixpkgs/pull/72312 as an example of commit message and history :)

File: pkgs/development/python-modules/pytest-bdd/default.nix (hunk 1 above)
Comment (jm2dev) [reply pkgs/development/python-modules/pytest-bdd/default.nix]: After running the following command:

```
find pkgs/development/python-modules -name default.nix | xargs grep doCheck | grep pytest
```

I suspect we really need that *doCheck = false*.

Comment (jonringer) [reply pkgs/development/python-modules/pytest-bdd/default.nix]: no... `doCheck = false;` will make it skip the checkPhase entirely

Here's how i would do this expression:
```
{ stdenv, buildPythonPackage, fetchFromGitHub
, execnet
, glob2
, Mako
, mock
, parse
, parse-type
, py
, pytest
, six
}:

buildPythonPackage rec {
  pname = "pytest-bdd";
  version = "3.2.1";

  # tests are not included in pypi tarball
  src = fetchFromGitHub {
    owner = "pytest-dev";
    repo = pname;
    rev = version;
    sha256 = "02y28l5h1m9grj54p681qvv7nrhd7ly9jkqdchyw4p0lnmcmnsrd";
  };

  propagatedBuildInputs = [ glob2 Mako parse parse-type py pytest six ];

  checkInputs = [ execnet mock pytest ];
  checkPhase = ''
    pytest
  '';

  meta = with stdenv.lib; {
    description = "BDD library for the py.test runner";
    homepage = https://github.com/pytest-dev/pytest-bdd;
    license = licenses.mit;
    maintainers = with maintainers; [ jm2dev ];
  };
}
```

Comment (jm2dev) [reply pkgs/development/python-modules/pytest-bdd/default.nix]: Thanks to that "fetchFromGitHub" now I'm aware of the available fetchers, and I suspect you  have plenty of experience with python pips.

Comment (jonringer) [reply pkgs/development/python-modules/pytest-bdd/default.nix]: I'm a co-owner of the nixpkgs python packages, only done a few hundred

**Comment by jm2dev** [comment]
Hi, do we have any updates on this PR?

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -3104,6 +3104,12 @@
     githubId = 2195834;
     name = "Jaakko Luttinen";
   };
+  jm2dev = {
+    email = "jomarcar@gmail.com";
+    github = "jm2dev";
+    githubId = 474643;
+    name = "José Miguel Martínez Carrasco";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix]: this should be a separate commit, look at https://github.com/NixOS/nixpkgs/pull/74389 as an example

Comment (jm2dev) [reply maintainers/maintainer-list.nix]: Done.

Comment (jonringer) [reply maintainers/maintainer-list.nix]: please reverse the order of the commits:

```
git rebase -i HEAD^^
# switch the two lines
# save
git push --force ...
```

Comment (jonringer) [reply maintainers/maintainer-list.nix]: otherwise checking out the package commit will fail to evaluate

Comment (jm2dev) [reply maintainers/maintainer-list.nix]: Output of `git log --oneline` after following your instructions (and pushing to gh):

```
0bced444e7f (HEAD -> pytest-bdd, origin/pytest-bdd) pythonPackages.pytest-bdd: init at 3.2.1
b1601d57ec9 maintainers: add jm2dev
```

though github web doesn't show this way.

Comment (jonringer) [reply maintainers/maintainer-list.nix]: ah, I looked at your branch, and it still showed it the other way. guess it is just the web view

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM

```
[5 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/72495
3 package were built:
python27Packages.pytest-bdd python37Packages.pytest-bdd python38Packages.pytest-bdd
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.pytest-bdd python37Packages.pytest-bdd python38Packages.pytest-bdd
